src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ConfigurationGeneralPage.xaml

```diff
@@ -0,0 +1,27 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information. -->
+<Rule Name="ConfigurationGeneral"
+      PageTemplate="generic"
+      Order="9999"
+      xmlns="http://schemas.microsoft.com/build/2009/properties" >
+
+  <!--
+  This "page" is not intended for display, and all properties contained within should be hidden.
+  
+  The properties below are used by visibility conditions defined in CPS and are expected, by convention,
+  to be present.
+  -->
+
+  <Rule.DataSource>
+    <DataSource Persistence="ProjectFile"
+                SourceOfDefaultValue="AfterContext"
+                HasConfigurationCondition="True" />
+  </Rule.DataSource>
+
+  <StringProperty Name="TargetFrameworkIdentifier" Visible="False" ReadOnly="True" />
+
+  <StringProperty Name="TargetFrameworkVersion" Visible="False" ReadOnly="True" />
+  
+  <StringProperty Name="LangVersion" Visible="False" ReadOnly="True" />
```

Comment:
I would expect `TargetFrameworkIdentifier` and `TargetFrameworkVersion` to be present in more or less all .NET projects, but I'm not sure about `LangVersion`. We should at least make sure it is applicable to C#, VB, and F#. Also, if we're going to have an intrinsic visibility expression that uses this we should make sure that it works with the values returned by those languages.

Reply:
`LangVersion` doesn't appear to be available to VB or F#. We will need an additional means to identify the project's language in visibility conditions, I suspect.